kolibri/plugins/coach/assets/src/modules/questionDetail/handlers.js

```diff
@@ -42,7 +42,17 @@ function showQuestionDetailView(params) {
     // of a combination of 'question_id:exercise_id'
     const baseExam = store.state.classSummary.examMap[quizId];
     promise = fetchExamWithContent(baseExam).then(({ exam }) => {
-      exerciseNodeId = exam.question_sources.find(source => source.item === questionId).exercise_id;
+      exam.question_sources.reduce((acc, source) => {
+        acc = [
+          ...acc,
+          ...source.questions.map(question => {
+            if (question.item === questionId) {
+              exerciseNodeId = question.exercise_id;
+            }
+          }),
+        ];
+        return acc;
+      }, []);
```

Comment:
This is working in that it is successfully setting the `exerciseNodeId` value to the proper value -- however, this could be done a bit more efficiently. 

As it works now, you're going to iterate over every question in the quiz, even if you found the correct question on the first iteration over the reduce callback function. In this case, it may be better consider another way forward. 

Typically when we call `reduce` on an array, we do it because we want to manipulate one shape of data into another altogether. In this case, `acc` is having it's value updated and manipulated as it goes through each iteration but then it's left behind unused.

---

Looking also at the code down below, there is another reducer which similarly is being used to find a particular value for the same `question` as your code here is.

You could look at how these scopes overlap and perhaps save some CPU cycles by doing some of this work above before the `promise = ...` line.

Thinking again about the reducer, you could consider using it to convert `exam.question_sources` (aka a list of objects) into a list of questions assigned to a new variable like `const allQuestions = exam.question_sources.reduce(...)`

That can then be used to `.find()` the matching question `allQuestions.find(q => q.item === questionId)` - then you can use that later on to get the `exercise_id` and generate the `title` below.